src/psyclone/transformations.py

```diff
@@ -563,6 +566,394 @@ def apply(self, node, options=None):
+            # Forward dependency positions are now computed for this region.
+            # Next we eliminate unneccessary dependencies. We loop over
+            # each dependence. For each dependence we loop over all other
+            # taskloops whose position is before that dependence. If that
+            # dependence is fulfilled by this dependence (i.e.
+            # dependence_position[this] < dependence_position[that]) then
+            # we remove that dependence. If this dependence is fulfilled
+            # by that dependence (i.e. dependence_position[that] <
+            # dependence_position[this]) then we remove this dependence.
+            # This assumes that taskloop_positions is in ascending order,
+            # which I believe to be true by construction.
+            for i, dep_pos in enumerate(dependence_position):
+                if dep_pos is not None:
+                    # Grab the position of the next dependence
+                    next_dependence = dep_pos
+                    # Loop over the other taskloops
+                    for j in range(i+1, len(dependence_position)):
+                        # If this taskloop happens after the next_dependence
+                        # then we can just stop.
+                        if taskloop_positions[j] >= next_dependence:
+                            break
+                        # If the jth taskloop has no dependency then continue
+                        if dependence_position[j] is None:
+                            continue
+                        # Check if next_dependence will satisfy the jth
+                        # taskloops dependency
+                        if next_dependence <= dependence_position[j]:
+                            dependence_position[j] = None
+                            continue
+                        # Check if the jth taskloops dependence will satisfy
+                        # the next_dependence
+                        if dependence_position[j] < next_dependence:
+                            dependence_position[i] = None
+                            # If it does then we can move to the next taskloop
+                            break
+            # dependence_position now contains only the required dependencies
+            # to satisfy the full superset of dependencies. We can loop over
+            # these by index, and if dependence_position[i] is not None then
+            # go to its forward dependency, and insert a TaskwaitDirective
+            # immediately before.
+            for i, dep_pos in enumerate(dependence_position):
+                if dep_pos is not None:
+                    forward_dep = OMPTaskwaitTrans.get_forward_dependence(
+                            taskloops[i], node)
+                    fdep_parent = forward_dep.parent
+                    # Find the position of the forward_dep in its parent's
+                    # children list
```

Comment:
I think you can just do `fdep_parent.children.index(forward_dep)`

Reply:
Changed